DEVELOPMENT.md

```diff
@@ -0,0 +1,115 @@
+type Logger interface {
+            ^
+cmd/drand-cli/cli_test.go:8:2: SA1019: "io/ioutil" has been deprecated since Go 1.16: As of Go 1.16, the same functionality is now provided by package io or package os, and those implementations should be preferred in new code. See the specific function documentation for details. (staticcheck)
+        "io/ioutil"
+        ^
+```
+
+- GoLand reports a few issues that might cause problems, especially around the nilness analyzer.
+- Some tests produce errors:
+
+TestLocalOrchestration
+```
+=== RUN   TestLocalOrchestration
+
+2022-10-03T22:49:32.804+0300    ERROR    127.0.0.1:44703.default.0    beacon/node.go:410        {"beacon_round": 6, "err_request": "rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp 127.0.0.1:41087: connect: connection refused\"", "from": "127.0.0.1:41087"}
+2022-10-03T22:49:32.804+0300    ERROR    127.0.0.1:44703.default.0.SyncManager.tryNode    beacon/sync_manager.go:358    unable_to_sync    {"with_peer": "127.0.0.1:37611", "err": "rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp 127.0.0.1:37611: connect: connection refused\""}
+```
+
+TestHTTPWaiting
+```
+=== RUN   TestHTTPWaiting
+2022-10-04T09:42:09.384+0300    INFO    http/server.go:127    New beacon handler registered    {"chainHash": "c5f1466e51ac94baa8a04fc9d5de8cd8f5e0cbb4985b70314e125d41f783b45c"}
+MOCK SERVER: emit round done
+2022-10-04T09:42:09.390+0300    WARN    http/server.go:390        {"http_server": "request in the future", "client": "[::1]:59544", "req": "%2Fc5f1466e51ac94baa8a04fc9d5de8cd8f5e0cbb4985b70314e125d41f783b45c%2Fpublic%2F1971"}
+    server_test.go:202: shouldn't be done. unexpected status: 404
+--- FAIL: TestHTTPWaiting (0.07s)
+```
+
+Generally running tests with `go test ./...`
+
+```
+2022-10-03T22:49:45.024+0300    DEBUG    127.0.0.1:37611.default.1.SyncManager.tryNode    beacon/sync_manager.go:434    sync canceled    {"source": "global", "err?": "context canceled"}
+2022-10-03T22:49:45.024+0300    DEBUG    127.0.0.1:37611.default.1.SyncManager    beacon/sync_manager.go:308    sync canceled early    {"source": "ctx", "err?": "context canceled"}
+2022-10-03T22:49:45.024+0300    INFO    127.0.0.1:41087.default.2    beacon/node.go:435    beacon handler stopped    {"time": "2022-10-03T22:49:45.024+0300"}
+2022-10-03T22:49:45.024+0300    INFO    net/client_grpc.go:245        {"grpc client": "chain sync", "error": "rpc error: code = Canceled desc = context canceled", "to": "127.0.0.1:41087"}
+2022-10-03T22:49:45.024+0300    INFO    127.0.0.1:44703.default.0    beacon/node.go:435    beacon handler stopped    {"time": "2022-10-03T22:49:45.024+0300"}
+2022-10-03T22:49:45.026+0300    ERROR    node/node_inprocess.go:312        {"drand": "failed to shutdown", "err": "rpc error: code = Unavailable desc = error reading from server: EOF"}
```

Comment:
Yeah, this is caused by the "control client <-> daemon" communication: when we stop the daemon, it stops the connection with the client and the client with exit with an EOF instead of knowing for sure that it worked and the daemon is shutdown.